src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+  health = 100;
 
-class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (this instanceof Carnivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health <= 0 && Animal.alive.indexOf(animal) >= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(animal), 1);
+      }
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+  }
```

Comment:
The `name` property is already set in the `Animal` constructor, so setting `this.name = name;` again in the `Carnivore` and `Herbivore` constructors is redundant and can be removed.